lib/rubocop/cop/style/double_pipe_equals.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for potential usage of the `||=` operator.
+      #
+      # @example
+      #   # bad
+      #   name = name ? name : 'Bozhidar'
+      #
+      #   # bad
+      #   name = if name
+      #            name
+      #          else
+      #            'Bozhidar'
+      #          end
+      #
+      #   # bad
+      #   unless name
+      #     name = 'Bozhidar'
+      #   end
+      #
+      #   # bad
+      #   name = 'Bozhidar' unless name
+      #
+      #   # good - set name to 'Bozhidar', only if it's nil or false
+      #   name ||= 'Bozhidar'
+      class DoublePipeEquals < Cop
+        MSG = 'Use the double pipe equals operator `||=` instead.'.freeze
+
+        def_node_matcher :ternary_assignment?, <<-PATTERN
+          (lvasgn _var
+            (if
+              (lvar _var)
+              (lvar _var)
+              _))
+        PATTERN
+
+        def_node_matcher :unless_assignment?, <<-PATTERN
+          (if
+            (send nil _var) nil
+            (lvasgn _var
+              _))
+        PATTERN
```

Comment:
This matcher is unnecessary. Because The result of parsing `foo = 1 unless foo` and `unless foo; foo = 1; end` is the same. However, `foo` needs to be predefined to be the same. Because `foo` is a local variable(`lvar` node) in the condition when `foo = 1 unless foo`, but `foo` is a method calling(`send` node) in the condition when `unless foo; foo = 1; end` if `foo` is not predefined.

For example: (`ruby-parse` is a utility of [parser](https://github.com/whitequark/parser) gem)

```bash
$ ruby-parse -e foo = 1 unless foo
(if
  (lvar :foo) nil
  (lvasgn :foo
    (int 1)))

$ ruby-parse -e unless foo; foo = 1; end
(if
  (send nil :foo) nil
  (lvasgn :foo
    (int 1)))

$ ruby-parse -e foo = nil; unless foo; foo = 1; end
(begin
  (lvasgn :foo
    (nil))
  (if
    (lvar :foo) nil
    (lvasgn :foo
      (int 1))))
```